packages/cli-core/src/package-manager-controller/package_manager_controller_base.ts

```diff
@@ -99,45 +99,28 @@ export abstract class PackageManagerControllerBase
 
   /**
    * initializeTsConfig - initializes a tsconfig.json file in the project root
+   *
+   * When changing this method, double check if a corresponding change is needed in the integration test setup in `setup_dir_as_esm_module.ts`.
    */
   async initializeTsConfig(targetDir: string) {
-    const tscArgs = [
-      'tsc',
-      '--init',
-      '--resolveJsonModule',
-      'true',
-      '--module',
-      'es2022',
-      '--moduleResolution',
-      'bundler',
-      '--target',
-      'es2022',
-    ];
-
-    await this.executeWithDebugLogger(
-      targetDir,
-      this.binaryRunner,
-      tscArgs,
-      this.execa,
-      { preferLocal: true, localDir: targetDir }
-    );
-
-    const pathsObj = {
-      // The path here is coupled with backend-function's generated typedef file path
-      '@env/*': ['../.amplify/function-env/*'],
+    const tsConfigTemplate = {
+      compilerOptions: {
+        target: 'es2022',
+        module: 'es2022',
+        moduleResolution: 'bundler',
+        resolveJsonModule: true,
+        // eslint-disable-next-line spellcheck/spell-checker
```

Comment:
I'd add to dictionary.

Reply:
The missing word is "interop" which isn't a word. Since this seems to be the first time we've needed it, I'm going to just ignore the line for now.

Reply:
https://en.wikipedia.org/wiki/Interop

Reply:
I've seen `interop` used as a shortcut in many independent places.

Reply:
https://en.wiktionary.org/wiki/interop

Reply:
okay I concede